etl/src/main/java/org/pih/hivmigration/etl/sql/ProcedureSetup.groovy

```diff
@@ -37,6 +38,7 @@ class ProcedureSetup extends SqlMigrator {
                     JOIN       hivmigration_encounters e ON o.source_encounter_id = e.source_encounter_id
                     LEFT JOIN  concept q ON q.uuid = o.concept_uuid
                     LEFT JOIN  concept a ON a.uuid = o.value_coded_uuid;
+                SET FOREIGN_KEY_CHECKS=1;    
```

Comment:
@cioan - don't turn off foreign key checking.  This would be bad practice.  It would be better to figure out what the dependency is and resolve it.  In this case, my guess is that groups need to be inserted prior to obs the refer to them, right?  So can we just change this procedure to do that?  Either by having 2 different insert statements one after the other (the first inserts where obs_group_id is null, the second inserts where obs_group_id is not null), or (if it works) have a single insert statement that just orders the select query in this way?

Reply:
I think @cioan asked on another thread, but interested how @brandones handled it with Lab Results.

Reply:
Like Mike says—the group needs to come before any obs that reference it.

Reply:
Thanks @mseaton , @mogoodrich and @brandones ! Of course, it was a bug in the statement I use to create the dispensing construct. The tmp_obs table and procedure Brandon created works fine for dispensing meds obs too. No need to turn off FOREIGN KEY checks. Thanks!